mackerel-plugin-h2o/lib/h2o.go

```diff
@@ -0,0 +1,306 @@
+		Label: "H2O Worker Threads",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "worker-threads", Label: "Worker Threads"},
+		},
+	},
+	"num-sessions": {
+		Label: "H2O Sessions",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "num-sessions", Label: "Sessions"},
+		},
+	},
+	"requests": {
+		Label: "H2O Requests",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "requests", Label: "In-flight Requests"},
+		},
+	},
+	"status-errors": {
+		Label: "H2O Status Errors",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "status-errors_400", Label: "Error 400", Diff: true},
+			{Name: "status-errors_403", Label: "Error 403", Diff: true},
+			{Name: "status-errors_404", Label: "Error 404", Diff: true},
+			{Name: "status-errors_405", Label: "Error 405", Diff: true},
+			{Name: "status-errors_416", Label: "Error 416", Diff: true},
+			{Name: "status-errors_417", Label: "Error 417", Diff: true},
+			{Name: "status-errors_500", Label: "Error 500", Diff: true},
+			{Name: "status-errors_502", Label: "Error 502", Diff: true},
+			{Name: "status-errors_503", Label: "Error 503", Diff: true},
+		},
+	},
+	"http2-errors": {
+		Label: "H2O HTTP2 Errors",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "http2-errors_protocol", Label: "Error protocol", Diff: true},
+			{Name: "http2-errors_internal", Label: "Error internal", Diff: true},
+			{Name: "http2-errors_flow-control", Label: "Error Flow Control", Diff: true},
+			{Name: "http2-errors_settings-timeout", Label: "Error Setting Timeout", Diff: true},
+			{Name: "http2-errors_frame-size", Label: "Error Frame Size", Diff: true},
+			{Name: "http2-errors_refused-stream", Label: "Error Refused Stream", Diff: true},
+			{Name: "http2-errors_cancel", Label: "Error Cancel", Diff: true},
+			{Name: "http2-errors_compression", Label: "Error Compression", Diff: true},
+			{Name: "http2-errors_connect", Label: "Error Connect", Diff: true},
+			{Name: "http2-errors_enhance-your-calm", Label: "Error Enhance Your Calm", Diff: true},
+			{Name: "http2-errors_inadequate-security", Label: "Error Inadequate Security", Diff: true},
+		},
+	},
+	"read-closed": {
+		Label: "H2O Read Closed",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "http2_read-closed", Label: "Read Closed", Diff: true},
+		},
+	},
+	"write-closed": {
+		Label: "H2O Write Closed",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "http2_write-closed", Label: "Write Closed", Diff: true},
+		},
+	},
+	"connect-time": {
+		Label: "H2O Connect Time",
+		Unit:  mp.UnitInteger,
+		Metrics: []mp.Metrics{
+			{Name: "connect-time-0", Label: "0 Percentile"},
+			{Name: "connect-time-25", Label: "25 Percentile"},
+			{Name: "connect-time-50", Label: "50 Percentile"},
+			{Name: "connect-time-75", Label: "75 Percentile"},
+			{Name: "connect-time-99", Label: "99 Percentile"},
+		},
+	},
+	"header-time": {
+		Label: "H2O Header Time",
+		Unit:  mp.UnitInteger,
```

Comment:
UnitFloat is better IMO.

Reply:
I thought the unit is in seconds but h2o outputs in `%lu` format. Is it in msec? nsec?

Reply:
its msec.

https://github.com/h2o/h2o/blob/master/lib/common/socket/evloop.c.h#L460-L465
https://github.com/h2o/h2o/blob/master/include/h2o/socket/uv-binding.h#L40-L43